From 52f93b59540470b6caf033db9eba33386e19621e Mon Sep 17 00:00:00 2001
From: Shen Yong <b00984@freescale.com>
Date: Tue, 16 Dec 2008 10:44:34 +0800
Subject: [PATCH] ENGR00100593 add mc13892 power off function

add mc13892 power off function callback for the power off process

Signed-off-by: Shen Yong<b00984@freescale.com>
---
 drivers/mxc/pmic/core/mc13892.c          |   11 +++++++++++
 include/asm-arm/arch-mxc/pmic_external.h |    1 +
 2 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/drivers/mxc/pmic/core/mc13892.c b/drivers/mxc/pmic/core/mc13892.c
index a5c52b1..c41f0c2 100644
--- a/drivers/mxc/pmic/core/mc13892.c
+++ b/drivers/mxc/pmic/core/mc13892.c
@@ -202,3 +202,14 @@ int pmic_event_mask(type_event event)
 
 	return ret;
 }
+
+void mc13892_power_off(void)
+{
+	unsigned int value;
+
+	pmic_read_reg(REG_POWER_CTL0, &value, 0xffffff);
+
+	value |= 0x000008;
+
+	pmic_write_reg(REG_POWER_CTL0, value, 0xffffff);
+}
diff --git a/include/asm-arm/arch-mxc/pmic_external.h b/include/asm-arm/arch-mxc/pmic_external.h
index 7ba9f26..1c1e5ec 100644
--- a/include/asm-arm/arch-mxc/pmic_external.h
+++ b/include/asm-arm/arch-mxc/pmic_external.h
@@ -959,6 +959,7 @@ int pmic_read(int reg_num, unsigned int *reg_val);
 int pmic_write(int reg_num, const unsigned int reg_val);
 void gpio_pmic_active(void);
 void pmic_event_list_init(void);
+void mc13892_power_off(void);
 
 #endif
 
-- 
1.5.4.4

